### PR TITLE
Next: Fixes and uiauto2 workloads

### PR DESCRIPTION
**Pull request by marcbonnici**

### CODE REVIEW

File: wa/utils/types.py [hunk 1]
```diff
@@ -586,3 +587,118 @@ def __new__(cls, name):
...
+        return dict.__contains__(self, self._encode(item))
+
+    def __iter__(self):
+        return iter((k, self._decode(v)) for (k, v) in self.items())
+
+    def iteritems(self):
+        return self.__iter__()
+
+    def get(self, name):
+        return self[name]
```
Comment (setrofim) [inline wa/utils/types.py]: This should be `return self._decode(dict.get(self, name))`. Note that `get()` and `__getitem__()` have different semantics when the key is not present.

File: wa/workloads/youtube/__init__.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        super(Youtube, self).__init__(device, **kwargs)
+        self.run_timeout = 300
+
+    def validate(self):
+        super(Youtube, self).validate()
+        self.gui.uiauto_params['video_source'] = self.video_source
+        self.gui.uiauto_params['search_term'] = self.search_term
+        # Make sure search term is set if video source is 'search'
+        if (self.video_source == 'search') and not self.search_term:
+            raise WorkloadError("Param 'search_term' must be specified when video source is 'search'")
```
Comment (setrofim) [inline wa/workloads/youtube/__init__.py]: Should be `ConfigError`

File: wa/workloads/youtube/uiauto/app/src/main/java/com/arm/wa/uiauto/youtube/UiAutomation.java [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    // // Pass the workload parameters, used for applaunch
+    // public void setWorkloadParameters(Bundle workload_parameters) {
+    //     parameters = workload_parameters;
+    // }
+
+    // public void setWorkloadParameters(Bundle parameters, String packageName, String packageID) {
+    //     this.parameters = parameters;
+    //     this.packageName = packageName;
+    //     this.packageID = packageID;
+    // }
```
Comment (setrofim) [inline wa/workloads/youtube/uiauto/app/src/main/java/com/arm/wa/uiauto/youtube/UiAutomation.java]: Should be removed.

File: wa/workloads/youtube/uiauto/app/src/main/java/com/arm/wa/uiauto/youtube/UiAutomation.java [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        ActionLogger logger = new ActionLogger(testTag + "_" + source, parameters);
+
+        if (SOURCE_SEARCH.equalsIgnoreCase(source)) {
+            clickUiObject(BY_DESC, "Search");
+            UiObject textField = getUiObjectByResourceId(packageID + "search_edit_text");
+            textField.setText(searchTerm);
+            mDevice.pressEnter();
+                // If a video exists whose title contains the exact search term, then play it
+                // Otherwise click the first video in the search results
+            UiObject thumbnail =
```
Comment (setrofim) [inline wa/workloads/youtube/uiauto/app/src/main/java/com/arm/wa/uiauto/youtube/UiAutomation.java]: Comment indentation

**Comment by marcbonnici** [comment]
Updated with requested changes.

**Pull request merged**